clients/src/main/java/org/apache/kafka/common/security/auth/AuthenticationContext.java

```diff
@@ -32,4 +34,9 @@
      * Address of the authenticated client
      */
     InetAddress clientAddress();
+
+    /**
+     * Name of the listener used for the connection
+     */
+    String listenerName();
```

Comment:
Can this be `null`? Maybe we should return `Optional<String>`. We would need to update the KIP too in that case.

Reply:
`AuthenticationContext` is only used on the server-side, so listener name should never be null.